### PR TITLE
feat: index referrers for image manifests

### PR DESCRIPTION
**Pull request by wangxiaoxuan273**
Part 8 of issue #21 
Part 1 of issue #39

### CODE REVIEW

**Review by shizhMSFT — APPROVED** [review]
LGTM with suggestions.

File: manifest/ocischema/manifest.go [hunk 1]
```diff
@@ -51,6 +51,10 @@ type Manifest struct {
 	// configuration.
 	Layers []distribution.Descriptor `json:"layers"`
 
+	// Subject specifies the descriptor of another manifest. This value is
+	// used by the referrers API.
+	Subject *distribution.Descriptor `json:"subject,omitempty"`
```
[the remark this answers is not in the document]
Comment (wangxiaoxuan273) [reply manifest/ocischema/manifest.go:56]: added `subject` in `Reference()` and changed the `verifyManifest` function accordingly.

File: registry/storage/ocimanifesthandler.go [hunk 2]
```diff
@@ -109,7 +118,7 @@ func (ms *ocischemaManifestHandler) verifyManifest(ctx context.Context, mnfst oc
 				}
 			}
 
-		case v1.MediaTypeImageManifest:
+		case v1.MediaTypeImageManifest, v1.MediaTypeArtifactManifest, v1.MediaTypeImageIndex, schema2.MediaTypeManifest:
```
Comment (shizhMSFT) [inline registry/storage/ocimanifesthandler.go:121]: how about docker manifest list?

Comment (wangxiaoxuan273) [reply registry/storage/ocimanifesthandler.go]: Added manifest list and schema1 manifest.

File: registry/storage/ocimanifesthandler.go [hunk 3]
```diff
@@ -118,7 +120,7 @@ func (ms *ocischemaManifestHandler) verifyManifest(ctx context.Context, mnfst oc
 				}
 			}
 
-		case v1.MediaTypeImageManifest, v1.MediaTypeArtifactManifest, v1.MediaTypeImageIndex, schema2.MediaTypeManifest:
+		case v1.MediaTypeImageManifest, v1.MediaTypeArtifactManifest, v1.MediaTypeImageIndex, schema1.MediaTypeManifest, schema2.MediaTypeManifest, manifestlist.MediaTypeManifestList:
```
Comment (shizhMSFT) [inline registry/storage/ocimanifesthandler.go:123]: We should remove `schema1` as it is deprecated.

Comment (wangxiaoxuan273) [reply registry/storage/ocimanifesthandler.go:123]: removed.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**